### PR TITLE
[wasm][debugger] Implement support to DebuggerDisplay attribute

### LINKED ISSUE #49442: [Blazor/WASM] Unable to enumerate over IEnumerable in the Watch Window
### Describe the bug
You cannot view the results of an IEnumerable in the watch window.

### To Reproduce
1. Create a new Blazor / WASM project in VS 2019
2. Add a new class called `Bar` using the code below
3. Update `FetchData.razor` using the code below.
4. Add a breakpoint after the `foo` variable is initialized
5. Add `foo` to the watch window
6. Expand the `foo` variable in the watch window
7. Expand the `Sequence` property (It's `IEnumerable<int>`)

### Expected Results
An option will appear in the watch window called "Results View" that will execute the enumerable and display the results

### Actual Results
No option exists to view the contents of the `IEnumerable` property.

#### WASM Output
image

#### .NET Core Output
image

### PR DESCRIPTION
**Pull request by thaystg**
image

### CODE REVIEW

**Review by lewing — APPROVED** [review]
Looks good other than snake case

File: src/mono/wasm/debugger/DebuggerTestSuite/CustomViewTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            var pause_location = await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_static_method ('[debugger-test] DebuggerTests.DebuggerCustomViewTest:run'); }, 1);",
+                "dotnet://debugger-test.dll/debugger-custom-view-test.cs",
+                bp.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp.Value["locations"][0]["columnNumber"].Value<int>(),
+                "run");
+
+            var locals = await GetProperties(pause_location["callFrames"][0]["callFrameId"].Value<string>());
+            CheckObject(locals, "a", "DebuggerTests.WithDisplayString", description:"Some one Value 2 End");
+			CheckObject(locals, "c", "DebuggerTests.DebuggerDisplayMethodTest", description: "First Int:32 Second Int:43");
```
Comment (lewing) [inline src/mono/wasm/debugger/DebuggerTestSuite/CustomViewTests.cs:30]:
```suggestion
			  CheckObject(locals, "c", "DebuggerTests.DebuggerDisplayMethodTest", description: "First Int:32 Second Int:43");
```

File: src/mono/wasm/debugger/tests/debugger-test/debugger-custom-view-test.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace DebuggerTests
+{
+    [DebuggerDisplay ("Some {Val1} Value {Val2} End")]
```
Comment (lewing) [inline src/mono/wasm/debugger/tests/debugger-test/debugger-custom-view-test.cs:11]:
```suggestion
    [DebuggerDisplay("Some {Val1} Value {Val2} End")]
```

File: src/mono/wasm/debugger/tests/debugger-test/debugger-custom-view-test.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    class WithToString
+    {
+        public override string ToString ()
+        {
+            return "SomeString";
+        }
+    }
+
+    [DebuggerTypeProxy (typeof(TheProxy))]
```
Comment (lewing) [inline src/mono/wasm/debugger/tests/debugger-test/debugger-custom-view-test.cs:27]:
```suggestion
    [DebuggerTypeProxy(typeof(TheProxy))]
```

File: src/mono/wasm/debugger/tests/debugger-test/debugger-custom-view-test.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        {
+            this.wp = wp;
+        }
+
+        public string Val2 {
+            get { return wp.Val1; }
+        }
+    }
+
+    [DebuggerDisplay ("{GetDebuggerDisplay(), nq}")]
```
Comment (lewing) [inline src/mono/wasm/debugger/tests/debugger-test/debugger-custom-view-test.cs:49]:
```suggestion
    [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
```

**Comment by thaystg** [comment]
> Looks good other than snake case

Fixed snake case.

**Comment by lewing** [comment]
fixes https://github.com/dotnet/runtime/issues/46859

**Review by lewing — APPROVED** [review]

**Pull request merged**